ksql-engine/src/main/java/io/confluent/ksql/util/ExecutorWithRetries.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import org.apache.kafka.common.errors.RetriableException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+
+public final class ExecutorWithRetries {
+
+  private static final int NUM_RETRIES = 5;
+  private static final int RETRY_BACKOFF_MS = 500;
+  private static final Logger log = LoggerFactory.getLogger(ExecutorWithRetries.class);
+
+  private ExecutorWithRetries() {
+  }
+
+  public enum RetryBehaviour {
```

Comment:
Maybe instead of an enum accept a collection of exceptions to retry on, or a Predicate<Exception> that the caller can implement to test what should be retried on (this would let you more cleanly handle exceptions from AdminClient wrapped in ExecutionException in KafkaTopicClientImpl).

Reply:
Good idea, but Is this something that could be left for a later PR if/when needed.

Reply:
Sure thing